tests/api2/test_108_catalog.py

```diff
@@ -0,0 +1,216 @@
+
+import os
+import pytest
+import sys
+apifolder = os.getcwd()
+sys.path.append(apifolder)
+from functions import GET, POST, DELETE, PUT, wait_on_job
+from auto_config import ha, scale, dev_test
+
+if dev_test:
+    reason = 'Skip for testing'
+else:
+    reason = 'Skipping test for HA' if ha else 'Skipping test for CORE'
+# comment pytestmark for development testing with --dev-test
+pytestmark = pytest.mark.skipif(ha or not scale or dev_test, reason=reason)
+
+official_repository = 'https://github.com/truenas/charts.git'
+truechart_repository = 'https://github.com/ericbsd/charts.git'
+official_chart = ['plex', 'nextcloud', 'minio', 'ix-chart', 'ipfs']
```

Comment:
This is okay but probably a better approach for OFFICIAL catalog would be to see what catalog items are present by directly looking up the git repository as these are bound to change with us adding more or maybe even removing some down the road.
So what can be done is basically that we can lookup the directories in this path `https://github.com/truenas/charts/tree/master/charts` and all of them are there own catalog items - which we can validate to ensure they are present in the payload as well provided by the middleware.

Reply:
Same could (ofc) be done with TrueCharts, with either the official repo or a "known good" stable fork which gets updated once in a while.

Reply:
> This is okay but probably a better approach for OFFICIAL catalog would be to see what catalog items are present by directly looking up the git repository as these are bound to change with us adding more or maybe even removing some down the road.
> So what can be done is basically that we can lookup the directories in this path `https://github.com/truenas/charts/tree/master/charts` and all of them are there own catalog items - which we can validate to ensure they are present in the payload as well provided by the middleware.

It can be done but it is a bit more complex unless there is a GITHUB REST API  call to list items in the derictory.

Reply:
done